homeassistant/components/melcloud/climate.py

```diff
@@ -0,0 +1,205 @@
+
+from pymelcloud import AtaDevice
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PRECISION_TENTHS, PRECISION_WHOLE, TEMP_CELSIUS
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util.temperature import convert as convert_temperature
+
+from .const import DOMAIN, HVAC_MODE_LOOKUP, HVAC_MODE_REVERSE_LOOKUP, TEMP_UNIT_LOOKUP
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+SCAN_INTERVAL = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+):
+    """Set up MelCloud device climate based on config_entry."""
+    mel_devices = hass.data[DOMAIN].get(entry.entry_id)
+    async_add_entities(
+        [
+            AtaDeviceClimate(mel_device)
+            for mel_device in mel_devices
+            if isinstance(mel_device.device, AtaDevice)
+        ],
+        True,
+    )
+
+
+class AtaDeviceClimate(ClimateDevice):
+    """Air-to-Air climate device."""
+
+    def __init__(self, device: AtaDevice, name=None):
+        """Initialize the climate."""
+        self._api = device
+        if name is None:
+            name = device.name
+
+        self._name = "{} {}".format(name, "HVAC")
+
+    @property
+    def unique_id(self) -> Optional[str]:
+        """Return a unique ID."""
+        return f"{self._api.device.serial}-{self._api.device.mac}-climate"
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return True
+
+    async def async_update(self):
+        """Update state from MELCloud."""
+        await self._api.async_update()
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return self._api.device_info
+
+    @property
+    def state(self) -> str:
+        """Return the current state."""
+        return self.hvac_mode
+
+    @property
+    def precision(self) -> float:
+        """Return the precision of the system."""
+        if self.hass.config.units.temperature_unit == TEMP_CELSIUS:
+            return PRECISION_TENTHS
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_UNIT_LOOKUP.get(self._api.device.temp_unit, TEMP_CELSIUS)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        mode = self._api.device.operation_mode
```

Comment:
You access `_api.device` in several places, maybe it's worth to store it in `self._device`?

Reply:
Fixed.